dali/pipeline/util/worker_thread.h

```diff
@@ -119,14 +119,14 @@ class WorkerThread {
     cv_.notify_one();
   }
 
-  inline void WaitForWork() {
+  inline void WaitForWork(bool rethrow_worker_errors = true) {
     std::unique_lock<std::mutex> lock(mutex_);
     while (!work_complete_) {
```

Comment:
Btw. it seems that calling WaitForWork while worker actually does something and then it raises the error, may deadlock as the worker thread does not set&notify completed_.